server/client.py

```diff
@@ -155,3 +160,74 @@ def send_message_to(self, msg):
 
         # send msg
         ONLINE_USERS[to].send(PayloadMessage(msg.cmd, [self.user.name], msg.payload))
+
+    @register_cmd(CMD_CREATE_CHAT)
+    @login_required
+    def create_chat(self, msg):
+        """
+        Create new chat and send new chat id back to the client
+        """
+        chat_name = msg.payload
+        db_session = session()
+        chat = Chat(chat_name, owner=self.user, users=[self.user,])
+        db_session.add(chat)
+        db_session.commit()
+        log.info("New chat '{}' has been created".format(chat))
+        self.send(PayloadMessage(CMD_INFO, [], str(chat.id)))
+
+    @register_cmd(CMD_ADD_CHAT_PARTICIPANT)
+    @login_required
+    def add_chat_participant(self, msg):
+        """
+        Add new participants to the chat
+        """
+        chat_id = msg.params[0]
+        participant_name = msg.params[1]
+        db_session = session()
+        chat = db_session.query(Chat).filter(Chat.id == chat_id).first()
+        if not chat:
+            raise InvalidChatID(chat)
+        if self.user.name not in [user.name for user in chat.users]:
+            raise InvalidChatID
+
+        chat = db_session.query(Chat).filter(Chat.id == chat_id).one()
+        chat.users.append(db_session.query(User).filter(User.name == participant_name).first())
+        db_session.add(chat)
+        db_session.commit()
+
+    @register_cmd(CMD_GET_CHATS)
+    @login_required
+    def get_chats(self, msg):
+        """
+        get all available chats for user
+        """
+        self.send(PayloadMessage(CMD_INFO, [], self.user.chats))
```

Comment:
The same about PayloadMessage for Info, we can create a defect for this and fix in separate CL.

Reply:
I will do it in separate CL